packages/language-server/server/completion/completion-item-kind.js

```diff
@@ -0,0 +1,74 @@
+const { CompletionItemKind } = require("vscode-languageserver");
+
+const FUNCTION_MATCHER = "=>";
+const CLASS_MATCHER = "class";
+const CONSTRUCTOR_MATCHER = "Constructor";
+
+const kinds = [
```

Comment:
It seems like the construction repeated from file to file, could we move it into the separated module?
Additionally. We can match the type by their constructors. 
1. Functions are instance of `FunctionType` or `GenericType` with `FunctionType` as `.subordinateType`.
2. Classes are `ObjectType` with `instanceType` or `GenericType` with  `ObjectType` (with `instanceType`) as `.subordinateType`.
3. Constructors are `ObjectType` with `constants.CONSTRUCTABLE` property.